### DISCUSSION TITLE
MOOSE Instation

### DISCUSSION THREAD

**Discussion by styyokuda** [idaholab/moose#24072] [Q&A Getting Started] [answered]
I have started to install MOOSE following the procedure.
Now I am at
"Upon restarting your terminal, you should see your prompt prefixed with (base). This indicates you are in the base environment, and Conda is ready for operation: $ (base) ~>"

But my terminal does not show "$ (base) ~>" . It shows just "~ %"
Please let me know if I can continue installation.
S. Thomas

**Comment by GiudGiud** [comment]
Hello

What do you get from `mamba activate moose` ?
then `mamba list | grep moose`

Guillaume

**Reply by styyokuda** [reply]
I did 

"mamba init
exit"

as the procedure instructs.

I think "mamba activate moose" is the later process.
I am sill in the process in the Conda MOOSE Environment.
I am just wondering it is OK to proceed with "~%" not the "$(base" prompt.
Please let me know.

S. Thomas

**Reply by GiudGiud** [reply]
mamba activate moose will not break anything

**Reply by GiudGiud** [reply]
you could mamba activate base instead

**Reply by styyokuda** [reply]
So, type "mamba activate moose" now?

Please let me know.

**Reply by GiudGiud** [reply]
`mamba activate base`
`mamba list`
what do you get

**Reply by styyokuda** [reply]
Is it OK to type "mamba init" again?
or it might conflict with some thing.

**Reply by styyokuda** [reply]
when type "mamba active base"
I got
"zsh: command not found: mamba"

Is it OK to type "mamba init" again?
or it might conflict with some thing.

**Reply by styyokuda** [reply]
Is it OK to re-start from the beginning again?
or it might conflict with some thing.
Please let me know.

**Reply by GiudGiud** [reply]
it s ok but it s not finding mamba so a step before that one failed

**Reply by styyokuda** [reply]
I did
export PATH=$HOME/mambaforge3/bin:$PATH
mamba init
The terminal shows:
"==> For changes to take effect, close and re-open your current shell. <==
Added mamba to /Users/d3k768/.bash_profile
==> For changes to take effect, close and re-open your current shell. <=="
Then, exit and re-start the terminal.  "mamba" does not work any more.
I am just wondering if I can keep moving to the next process of
"conda config --add channels https://conda.software.inl.gov/public" with the "~%" prompt in the terminal.
please let me know.

**Reply by GiudGiud** [reply]
hello

no you cant because when you'll type `conda` it likely wont find it. 

export PATH=$HOME/mambaforge3/bin:$PATH
is this correct?
what does `ls $HOME/mambaforge3/bin` give you?

**Reply by styyokuda** [reply]
I did
"export $HOME/mambaforge3/bin"
Then, I got
"export: not valid in this context: ~/mambaforge3/bin"

**Reply by styyokuda** [reply]
Sorry I misunderstood.
with "ls $HOME/mambaforge3/bin"
I see a bunch of file there.

**Reply by GiudGiud** [reply]
the export needs to succeed.
why is this not working

**Reply by GiudGiud** [reply]
maybe you need to write the explicit full path, not ~

**Reply by styyokuda** [reply]
The exact terminal prompt is

"d3k768@WE47574 ~ % "

**Reply by GiudGiud** [reply]
what does `echo $HOME` return?

**Reply by styyokuda** [reply]
Yes as
"/Users/d3k768"

**Reply by styyokuda** [reply]
This is MacBOOK Pro with Apple M2 Max chip.
It is a brand new. I just got it yesterday.

**Reply by GiudGiud** [reply]
if you write down the full path to mamba does it work?

**Reply by styyokuda** [reply]
I don't understand  "full path to mamba"
Please help.

**Reply by GiudGiud** [reply]
the full path to mamba is $HOME/mambaforge3/bin/ + mamba  I think

**Reply by styyokuda** [reply]
I see the command "mamba" in/ Users/d3k768/mambaforge3/bin.
So, are you asking  to run / Users/d3k768/mambaforge3/bin/mamba activate moose?

**Reply by GiudGiud** [reply]
yes

**Reply by GiudGiud** [reply]
or just `base` for now

**Reply by styyokuda** [reply]
I typed
"/Users/d3k768/mambaforge3/bin/mamba activate moose"
then I got
"Run 'mamba init' to be able to run mamba activate/deactivate
and start a new shell session. Or use conda to activate/deactivate."

Does it tell you anything?

**Reply by styyokuda** [reply]
Also I typed
"/Users/d3k768/mambaforge3/bin/mamba activate base "
then I got
"Run 'mamba init' to be able to run mamba activate/deactivate
and start a new shell session. Or use conda to activate/deactivate."

**Reply by GiudGiud** [reply]
Ok so you can get going with just replacing mamba by the full path to mamba for now

What does echo $PATH return? We should figure out why we can’t find mamba on your path

**Reply by GiudGiud** [reply]
Did you su to an admin or sudo at any point during the process ?

**Reply by styyokuda** [reply]
No, I did not use su nor sudo.
echo $PATH gives,/Users/d3k768/mambaforge3/bin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin

**Reply by GiudGiud** [reply]
looks fine.
Did the init work?

**Reply by styyokuda** [reply]
mamba init gives
no change     /Users/d3k768/mambaforge3/condabin/conda
no change     /Users/d3k768/mambaforge3/bin/conda
no change     /Users/d3k768/mambaforge3/bin/conda-env
no change     /Users/d3k768/mambaforge3/bin/activate
no change     /Users/d3k768/mambaforge3/bin/deactivate
no change     /Users/d3k768/mambaforge3/etc/profile.d/conda.sh
no change     /Users/d3k768/mambaforge3/etc/fish/conf.d/conda.fish
no change     /Users/d3k768/mambaforge3/shell/condabin/Conda.psm1
no change     /Users/d3k768/mambaforge3/shell/condabin/conda-hook.ps1
no change     /Users/d3k768/mambaforge3/lib/python3.10/site-packages/xontrib/conda.xsh
no change     /Users/d3k768/mambaforge3/etc/profile.d/conda.csh
modified      /Users/d3k768/.bash_profile

==> For changes to take effect, close and re-open your current shell. <==

Added mamba to /Users/d3k768/.bash_profile

==> For changes to take effect, close and re-open your current shell. <==

d3k768@WE47574 ~ % 

and exit and restart it, I get
d3k768@WE47574 ~ % 
Can I move to the next process with it?

**Reply by GiudGiud** [reply]
what do you get when you try to activate

**Reply by styyokuda** [reply]
After exit,
mamba activate base  gives[
zsh: command not found: mamba

still the same problem ...

**Reply by GiudGiud** [reply]
what does ls /Users/d3k768/mambaforge3/bin give again?

**Reply by styyokuda** [reply]
ls /Users/d3k768/mambaforge3/bin
lists a bunch of files.

**Reply by GiudGiud** [reply]
does that include mamba?

**Reply by milljm** [reply]
Hmmm, so `mamba init` believe you are operating within a BASH shell and not the ZSH you are actually in.

Not sure why this is the case. Based on what you've shown us, can you add the following to the end of your `~/.zshrc` file:

```bash
# >>> conda initialize >>>
# !! Contents within this block are managed by 'conda init' !!
__conda_setup="$('/Users/d3k768/mambaforge3/bin/conda' 'shell.zsh' 'hook' 2> /dev/null)"
if [ $? -eq 0 ]; then
    eval "$__conda_setup"
else
    if [ -f "/Users/d3k768/mambaforge3/etc/profile.d/conda.sh" ]; then
        . "/Users/d3k768/mambaforge3/etc/profile.d/conda.sh"
    else
        export PATH="/Users/d3k768/mambaforge3/bin:$PATH"
    fi
fi
unset __conda_setup

if [ -f "/Users/d3k768/mambaforge3/etc/profile.d/mamba.sh" ]; then
    . "/Users/d3k768/mambaforge3/etc/profile.d/mamba.sh"
fi
# <<< conda initialize <<<

```

My advice would be to use `vim` for this. Not sure how familiar you are with VIM, so I'll try to give some pointers:
```bash
vim ~/.zshrc
```

_Note: where you see me use `+` keys below, don't actually strike the plus key. It denotes that you should hold down both keys, starting with the first one_

Using the arrow keys, arrow down to the end of the file, Then press `o` on the keyboard to add a new line, and to enter 'input' mode. Paste the above code block `command+v`, then press the following key sequence to save and exit vim: `shift+:`, then press `wq` then press enter.

`shift+: ` enters command mode. Think of it like clicking on "File" on the top left of a GUI program :). `wq` is actually two commands; `w` write the file. `q` quits.

Once all this is done, close and open your terminals. Hopefully then you will see `(base)` within your prompt, and you can continue to "installing MOOSE" on our instructions page.

**Reply by styyokuda** [reply]
I don't think I have the .zshrc file.
see below.

d3k768@WE47574 ~ % more ~/.zshrc       
/Users/d3k768/.zshrc: No such file or directory

**Reply by GiudGiud** [reply]
what does
`ps -p $$` return

**Reply by styyokuda** [reply]
[d3k768@WE47574 ~ % ps -p $$
  PID TTY           TIME CMD
 7961 ttys001    0:00.06 -zsh

**Reply by milljm** [reply]
Then you proceed with my instructions. By doing so, you will create the file.

**Reply by styyokuda** [reply]
Is the following is what I need?

(base) d3k768@WE47574 ~ %

**Reply by styyokuda** [reply]
with 
`mamba create -n moose moose-tools moose-libmesh`
I got the following error message:
```
# >>>>>>>>>>>>>>>>>>>>>> ERROR REPORT <<<<<<<<<<<<<<<<<<<<<<

    Traceback (most recent call last):
      File "/Users/d3k768/mambaforge3/lib/python3.10/site-packages/conda/exceptions.py", line 1124, in __call__
        return func(*args, **kwargs)
      File "/Users/d3k768/mambaforge3/lib/python3.10/site-packages/mamba/mamba.py", line 941, in exception_converter
        raise e
      File "/Users/d3k768/mambaforge3/lib/python3.10/site-packages/mamba/mamba.py", line 934, in exception_converter
        exit_code = _wrapped_main(*args, **kwargs)
      File "/Users/d3k768/mambaforge3/lib/python3.10/site-packages/mamba/mamba.py", line 892, in _wrapped_main
        result = do_call(parsed_args, p)
      File "/Users/d3k768/mambaforge3/lib/python3.10/site-packages/mamba/mamba.py", line 758, in do_call
        exit_code = create(args, parser)
      File "/Users/d3k768/mambaforge3/lib/python3.10/site-packages/mamba/mamba.py", line 632, in create
        return install(args, parser, "create")
      File "/Users/d3k768/mambaforge3/lib/python3.10/site-packages/mamba/mamba.py", line 499, in install
        index = load_channels(pool, channels, repos)
      File "/Users/d3k768/mambaforge3/lib/python3.10/site-packages/mamba/utils.py", line 129, in load_channels
        index = get_index(
      File "/Users/d3k768/mambaforge3/lib/python3.10/site-packages/mamba/utils.py", line 110, in get_index
        is_downloaded = dlist.download(api.MAMBA_DOWNLOAD_FAILFAST)
    RuntimeError: Download error (35) SSL connect error [https://conda.software.inl.gov/public/noarch/repodata.json]
    OpenSSL/3.1.0: error:0A000152:SSL routines::unsafe legacy renegotiation disabled

`$ /Users/d3k768/mambaforge3/bin/mamba create -n moose moose-tools moose-libmesh`

  environment variables:
                 CIO_TEST=<not set>
        CONDA_DEFAULT_ENV=base
                CONDA_EXE=/Users/d3k768/mambaforge3/bin/conda
             CONDA_PREFIX=/Users/d3k768/mambaforge3
    CONDA_PROMPT_MODIFIER=(base)
         CONDA_PYTHON_EXE=/Users/d3k768/mambaforge3/bin/python
               CONDA_ROOT=/Users/d3k768/mambaforge3
              CONDA_SHLVL=1
           CURL_CA_BUNDLE=<not set>
               LD_PRELOAD=<not set>
                     PATH=/Users/d3k768/mambaforge3/bin:/Users/d3k768/mambaforge3/condabin:/usr/
                          local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:
                          /var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/
                          bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bi
                          n:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appl
                          einternal/bin
       REQUESTS_CA_BUNDLE=<not set>
            SSL_CERT_FILE=<not set>

     active environment : base
    active env location : /Users/d3k768/mambaforge3
            shell level : 1
       user config file : /Users/d3k768/.condarc
 populated config files : /Users/d3k768/mambaforge3/.condarc
                          /Users/d3k768/.condarc
          conda version : 23.1.0
    conda-build version : not installed
         python version : 3.10.10.final.0
       virtual packages : __archspec=1=arm64
                          __osx=13.3=0
                          __unix=0=0
       base environment : /Users/d3k768/mambaforge3  (writable)
      conda av data dir : /Users/d3k768/mambaforge3/etc/conda
  conda av metadata url : None
           channel URLs : https://conda.software.inl.gov/public/osx-arm64
                          https://conda.software.inl.gov/public/noarch
                          https://conda.anaconda.org/conda-forge/osx-arm64
                          https://conda.anaconda.org/conda-forge/noarch
          package cache : /Users/d3k768/mambaforge3/pkgs
                          /Users/d3k768/.conda/pkgs
       envs directories : /Users/d3k768/mambaforge3/envs
                          /Users/d3k768/.conda/envs
               platform : osx-arm64
             user-agent : conda/23.1.0 requests/2.28.2 CPython/3.10.10 Darwin/22.4.0 OSX/13.3
                UID:GID : 501:20
             netrc file : None
           offline mode : False

An unexpected error has occurred. Conda has prepared the above report.
```

**Reply by GiudGiud** [reply]
You can try enabling unsafe SSH, **at your own risk**
`conda config --set ssl_verify false`

**Reply by milljm** [reply]
The only way for this to come up, is if you are trying to access our site via behind a companies firewall/proxy that is performing some form of man-in-the-middle packet inspection.

```pre
    RuntimeError: Download error (35) SSL connect error [https://conda.software.inl.gov/public/noarch/repodata.json]
    OpenSSL/3.1.0: error:0A000152:SSL routines::unsafe legacy renegotiation disabled
```

As @GiudGiud mentioned, you can tell Conda to trust "everything" by telling it not to verify SSL certificates. Your call!

**Reply by styyokuda** [reply]
I did
conda config --set ssl_verify false

Can I move to the next step?

**Reply by GiudGiud** [reply]
yes, try installing the moose environment please

**Reply by styyokuda** [reply]
I got the following.
(base) d3k768@WE47574 ~ % mamba activate moose

EnvironmentNameNotFound: Could not find conda environment: moose
You can list all discoverable environments with `conda info --envs`.

**Reply by GiudGiud** [reply]
you have to install first, the step right before
mamba create -n moose moose-tools moose-libmesh

**Reply by styyokuda** [reply]
Am I supposed to repeat the process of
conda config --add channels https://conda.software.inl.gov/public
mamba create -n moose moose-tools moose-libmesh
conda config --set ssl_verify false
mamba activate moose

Please let me know.

**Reply by GiudGiud** [reply]
Just run the second command

**Reply by milljm** [reply]
```bash
conda config --set ssl_verify false
conda config --add channels https://conda.software.inl.gov/public
mamba create -n moose moose-tools moose-libmesh
mamba activate moose
```
Set the SSL verify flag first... Then do the rest.

**Reply by styyokuda** [reply]
I did
(base) d3k768@WE47574 ~ % conda config --set ssl_verify false
(base) d3k768@WE47574 ~ % conda config --add channels https://conda.software.inl.gov/public
Warning: 'https://conda.software.inl.gov/public' already in 'channels' list, moving to the top
(base) d3k768@WE47574 ~ % mamba create -n moose moose-tools moose-libmesh
Then, I got the same error report as the one  before.

**Reply by milljm** [reply]
Hmmm you might need to downgrade your openSSL package in the base environment... In order to do that, we first need to remove our channel (otherwise the same error will occur). Please try the following:

```bash
conda config --remove channels https://conda.software.inl.gov/public
mamba install openssl=1
conda config --add channels https://conda.software.inl.gov/public
mamba create -n moose moose-tools moose-libmesh
```

For the record, are you operating within some corporate firewall location? The error you are receiving suggests you are being "watched". If you are using your corporate firewall: then this is probably fine. If you're at home, using your own internet: could be bad.

**Reply by styyokuda** [reply]
I did
(base) d3k768@WE47574 ~ % conda config --remove channels https://conda.software.inl.gov/public
(base) d3k768@WE47574 ~ % mamba install openssl=1

Then, I got the same error report as the one by mamba create -n moose moose-tools moose-libmesh

**Reply by milljm** [reply]
Can perform a `mamba list`, and display the contents?

**Reply by styyokuda** [reply]
```pre
(base) d3k768@WE47574 ~ % mamba list
# packages in environment at /Users/d3k768/mambaforge3:
#
# Name                    Version                   Build  Channel
brotlipy                  0.7.0           py310h8e9501a_1005    conda-forge
bzip2                     1.0.8                h3422bc3_4    conda-forge
c-ares                    1.18.1               h3422bc3_0    conda-forge
ca-certificates           2022.12.7            h4653dfc_0    conda-forge
certifi                   2022.12.7          pyhd8ed1ab_0    conda-forge
cffi                      1.15.1          py310h2399d43_3    conda-forge
charset-normalizer        3.1.0              pyhd8ed1ab_0    conda-forge
colorama                  0.4.6              pyhd8ed1ab_0    conda-forge
conda                     23.1.0          py310hbe9552e_0    conda-forge
conda-package-handling    2.0.2              pyh38be061_0    conda-forge
conda-package-streaming   0.7.0              pyhd8ed1ab_1    conda-forge
cryptography              40.0.1          py310hfc83b78_0    conda-forge
fmt                       9.1.0                hffc8910_0    conda-forge
icu                       72.1                 he12128b_0    conda-forge
idna                      3.4                pyhd8ed1ab_0    conda-forge
krb5                      1.20.1               h69eda48_0    conda-forge
libarchive                3.6.2                h83f22c9_0    conda-forge
libcurl                   7.88.1               h9049daf_1    conda-forge
libcxx                    16.0.0               h75e25f2_0    conda-forge
libedit                   3.1.20191231         hc8eb9b7_2    conda-forge
libev                     4.33                 h642e427_1    conda-forge
libffi                    3.4.2                h3422bc3_5    conda-forge
libiconv                  1.17                 he4db4b2_0    conda-forge
libmamba                  1.4.1                h7d1d596_0    conda-forge
libmambapy                1.4.1           py310h34b6e76_0    conda-forge
libnghttp2                1.52.0               hae82a92_0    conda-forge
libsolv                   0.7.23               hb5ab8b9_0    conda-forge
libsqlite                 3.40.0               h76d750c_0    conda-forge
libssh2                   1.10.0               h7a5bd25_3    conda-forge
libxml2                   2.10.3               h2aff0a6_6    conda-forge
libzlib                   1.2.13               h03a7124_4    conda-forge
lz4-c                     1.9.4                hb7217d7_0    conda-forge
lzo                       2.10              h642e427_1000    conda-forge
mamba                     1.4.1           py310ha5d4528_0    conda-forge
ncurses                   6.3                  h07bb92c_1    conda-forge
openssl                   3.1.0                h03a7124_0    conda-forge
pip                       23.0.1             pyhd8ed1ab_0    conda-forge
pluggy                    1.0.0              pyhd8ed1ab_5    conda-forge
pybind11-abi              4                    hd8ed1ab_3    conda-forge
pycosat                   0.6.4           py310h8e9501a_1    conda-forge
pycparser                 2.21               pyhd8ed1ab_0    conda-forge
pyopenssl                 23.1.1             pyhd8ed1ab_0    conda-forge
pysocks                   1.7.1              pyha2e5f31_6    conda-forge
python                    3.10.10         h3ba56d0_0_cpython    conda-forge
python_abi                3.10                    3_cp310    conda-forge
readline                  8.2                  h92ec313_1    conda-forge
reproc                    14.2.4               h1a8c8d9_0    conda-forge
reproc-cpp                14.2.4               hb7217d7_0    conda-forge
requests                  2.28.2             pyhd8ed1ab_1    conda-forge
ruamel.yaml               0.17.21         py310h8e9501a_3    conda-forge
ruamel.yaml.clib          0.2.7           py310h8e9501a_1    conda-forge
setuptools                65.6.3             pyhd8ed1ab_0    conda-forge
tk                        8.6.12               he1e0b03_0    conda-forge
toolz                     0.12.0             pyhd8ed1ab_0    conda-forge
tqdm                      4.65.0             pyhd8ed1ab_1    conda-forge
tzdata                    2023c                h71feb2d_0    conda-forge
urllib3                   1.26.15            pyhd8ed1ab_0    conda-forge
wheel                     0.40.0             pyhd8ed1ab_0    conda-forge
xz                        5.2.6                h57fd34a_0    conda-forge
yaml-cpp                  0.7.0                hb7217d7_2    conda-forge
zstandard                 0.19.0          py310had9512b_1    conda-forge
zstd                      1.5.2                hf913c23_6    conda-forge
(base) d3k768@WE47574 ~ % 
```

**Reply by GiudGiud** [reply]
@styyokuda please format the pasted snippets with triple quotes \`\`\` before and after the text, it looks better

**Reply by milljm** [reply]
```pre
ncurses                   6.3                  h07bb92c_1    conda-forge
openssl                   3.1.0                h03a7124_0    conda-forge <-----
pip                       23.0.1             pyhd8ed1ab_0    conda-forge
```
This is still version 3. Can you please perform the openssl=1 install I asked earlier? And because you re-added our public channel, you'll need to remove it first (basically do exactly what I asked _again_)

```bash
conda config --remove channels https://conda.software.inl.gov/public
mamba install openssl=1
conda config --add channels https://conda.software.inl.gov/public
mamba create -n moose moose-tools moose-libmesh
```

Please stop when/if you receive an error, and do not proceed. We need to know if for example:
```bash
mamba install openssl=1
```
fails... It if does, there is no need to proceed. As the end result will always end if failure.

**Reply by styyokuda** [reply]
(base) d3k768@WE47574 ~ % conda config --remove channels https://conda.software.inl.gov/public
works.
But 
```pre
(base) d3k768@WE47574 ~ % mamba install openssl=1
# >>>>>>>>>>>>>>>>>>>>>> ERROR REPORT <<<<<<<<<<<<<<<<<<<<<<

    Traceback (most recent call last):
      File "/Users/d3k768/mambaforge3/lib/python3.10/site-packages/conda/exceptions.py", line 1124, in __call__
        return func(*args, **kwargs)
      File "/Users/d3k768/mambaforge3/lib/python3.10/site-packages/mamba/mamba.py", line 941, in exception_converter
        raise e
      File "/Users/d3k768/mambaforge3/lib/python3.10/site-packages/mamba/mamba.py", line 934, in exception_converter
        exit_code = _wrapped_main(*args, **kwargs)
      File "/Users/d3k768/mambaforge3/lib/python3.10/site-packages/mamba/mamba.py", line 892, in _wrapped_main
        result = do_call(parsed_args, p)
      File "/Users/d3k768/mambaforge3/lib/python3.10/site-packages/mamba/mamba.py", line 754, in do_call
        exit_code = install(args, parser, "install")
      File "/Users/d3k768/mambaforge3/lib/python3.10/site-packages/mamba/mamba.py", line 499, in install
        index = load_channels(pool, channels, repos)
      File "/Users/d3k768/mambaforge3/lib/python3.10/site-packages/mamba/utils.py", line 129, in load_channels
        index = get_index(
      File "/Users/d3k768/mambaforge3/lib/python3.10/site-packages/mamba/utils.py", line 110, in get_index
        is_downloaded = dlist.download(api.MAMBA_DOWNLOAD_FAILFAST)
    RuntimeError: Download error (35) SSL connect error [https://conda.anaconda.org/conda-forge/noarch/repodata.json]
    OpenSSL/3.1.0: error:0A000152:SSL routines::unsafe legacy renegotiation disabled

`$ /Users/d3k768/mambaforge3/bin/mamba install openssl=1`

  environment variables:
                 CIO_TEST=<not set>
        CONDA_DEFAULT_ENV=base
                CONDA_EXE=/Users/d3k768/mambaforge3/bin/conda
             CONDA_PREFIX=/Users/d3k768/mambaforge3
    CONDA_PROMPT_MODIFIER=(base)
         CONDA_PYTHON_EXE=/Users/d3k768/mambaforge3/bin/python
               CONDA_ROOT=/Users/d3k768/mambaforge3
              CONDA_SHLVL=1
           CURL_CA_BUNDLE=<not set>
               LD_PRELOAD=<not set>
                     PATH=/Users/d3k768/mambaforge3/bin:/Users/d3k768/mambaforge3/condabin:/usr/
                          local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:
                          /var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/
                          bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bi
                          n:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appl
                          einternal/bin
       REQUESTS_CA_BUNDLE=<not set>
            SSL_CERT_FILE=<not set>

     active environment : base
    active env location : /Users/d3k768/mambaforge3
            shell level : 1
       user config file : /Users/d3k768/.condarc
 populated config files : /Users/d3k768/mambaforge3/.condarc
                          /Users/d3k768/.condarc
          conda version : 23.1.0
    conda-build version : not installed
         python version : 3.10.10.final.0
       virtual packages : __archspec=1=arm64
                          __osx=13.3=0
                          __unix=0=0
       base environment : /Users/d3k768/mambaforge3  (writable)
      conda av data dir : /Users/d3k768/mambaforge3/etc/conda
  conda av metadata url : None
           channel URLs : https://conda.anaconda.org/conda-forge/osx-arm64
                          https://conda.anaconda.org/conda-forge/noarch
          package cache : /Users/d3k768/mambaforge3/pkgs
                          /Users/d3k768/.conda/pkgs
       envs directories : /Users/d3k768/mambaforge3/envs
                          /Users/d3k768/.conda/envs
               platform : osx-arm64
             user-agent : conda/23.1.0 requests/2.28.2 CPython/3.10.10 Darwin/22.4.0 OSX/13.3
                UID:GID : 501:20
             netrc file : None
           offline mode : False
```

**Reply by milljm** [reply]
This is unfortunate. It would seem you are also "not allowed" to connect to the Conda-Forge channel without first allowing your traffic to this site to be inspected by a third party.

There is nothing we are going to be able to do to remedy this. You'll need to get in touch with whomever your network administrator is. Explain to them that OpenSSL 3 prevents man-in-the-middle inspections (something your proxy/firewall is trying to do). Personally, this is a good thing (privacy). But perhaps your network does not allow it.

What is interesting, is that you're probably allowed to click on the following link: https://conda.anaconda.org/conda-forge/noarch/repodata.json. This would mean your web browser is properly configured for handling such requests through your Proxy server (which _is_ performing packet inspection).

Conda, has no such setting.

**Reply by milljm** [reply]
You _can try_ to download an older version of Mambaforge... one that came with OpenSSL 1.x.

You will of course need to delete your current installed Mambaforge before hand:

```bash
rm -rf ~/mambaforge3
```

Download and install this older version which installs `OpenSSL 1.1.1o`:
```bash
curl -L -O https://github.com/conda-forge/miniforge/releases/download/4.12.0-2/Mambaforge-Darwin-arm64.sh
sh Mambaforge-Darwin-arm64.sh -b -p ~/mambaforge3
```
Close all your terminal windows, open them back up again (the information we added to your `~/.zshrc` should still work, and bring you to a `(base)` prefixed prompt).

From here, you should be able to install our Conda packages:
```bash
conda config --add channels https://conda.software.inl.gov/public
mamba create -n moose moose-tools moose-libmesh openssl=1
mamba activate moose
```
The only caveat is that you will continuously need to specify `openssl=1` when ever you run `install`, `create`, or `update` commands.

Basically, Conda will attempt to _always_ upgrade OpenSSL 1.x to 3.x. And once that happens, **game over**. You'll need to delete your installation of Mambaforge, and start over (there is no downgrading, because downgrading requires a connection to do so, and a connection is what you'll not be able to do).

**Reply by styyokuda** [reply, marked as answer]
Great MOOSE Developers,
I had no problems with MOOSE Installation on my old MacBook Pro with WiFi.  But this brand new MacBook Pro needs to be connected through a cable .  It is working now.  Thank you very, very much for your time and great patience with me. I love MOOSE. S. Thomas

**Reply by milljm** [reply]
I've ran into MPI issues with Ethernet vs WiFi, but this is new. I am glad you got it sorted! And thank you for reporting back! I'll be sure to remember switching network methods as a possible troubleshooting step.

**Comment by zaheen1258** [comment]
Hello @styyokuda, So I'm facing the same problem with conda installation in a brand new MacBook Pro with M2 Pro chip. I've already tried all the steps above like creating a .zshrc file but couldn't reach "(base)xxxx@xxx" step. So how did you actually overcame this problem? Thanks in advance.

**Reply by milljm** [reply]
Same issue as in not being able to get a `(base)` environment after install?

If I'm reading you correctly, I've run into the same issue with new laptops issues by our organization which "profiles" our machines. Perhaps the same issue exists for you; this is just a guess:

```bash
❯ echo $0
-zsh
```

While this may say ZSH is my shell, somewhere/somehow, the Conda installer detects you might still be using BASH. This is even after running the hint command supplied by Apple when starting in a BASH shell:

```bash
The default interactive shell is now zsh.
To update your account to use zsh, please run `chsh -s /bin/zsh`.
For more details, please visit https://support.apple.com/kb/HT208050.
```

Following the above directions _does not seem to fix the issue_. Unless you also make the correct modifications to your `~/.zshrc` file manually, or, by heading into "System Settings", "Users & Groups", and making the "default shell" modification there (involves right-click advanced options clickery, which I no longer have access to on my corporate managed machine).

If you are unable to make the necessary changes using my "System Settings" instructions, then you are left with creating the necessary `~/.zshrc` changes manually.

```bash
# >>> conda initialize >>>
# !! Contents within this block are managed by 'conda init' !!
__conda_setup="$('/Users/d3k768/mambaforge3/bin/conda' 'shell.zsh' 'hook' 2> /dev/null)"
                         ^^^^^ this                           ^^^ and this
if [ $? -eq 0 ]; then
    eval "$__conda_setup"
else
    if [ -f "/Users/d3k768/mambaforge3/etc/profile.d/conda.sh" ]; then
        . "/Users/d3k768/mambaforge3/etc/profile.d/conda.sh"
    else
        export PATH="/Users/d3k768/mambaforge3/bin:$PATH"
    fi
fi
unset __conda_setup

if [ -f "/Users/d3k768/mambaforge3/etc/profile.d/mamba.sh" ]; then
    . "/Users/d3k768/mambaforge3/etc/profile.d/mamba.sh"
fi
# <<< conda initialize <<<
```

Where `this`, `and this` need to change accordingly, throughout the file.

If the above has no effect, try copy/pasting the initialization contents into a new file and sourcing it:

`cat`'ing the file as an example:
```bash
❯  cat mamba_init
# >>> conda initialize >>>
# !! Contents within this block are managed by 'conda init' !!
__conda_setup="$('/Users/YOURUSERNAME/mambaforge3/bin/conda' 'shell.zsh' 'hook' 2> /dev/null)"
if [ $? -eq 0 ]; then
    eval "$__conda_setup"
else
    if [ -f "/Users/YOURUSERNAME/mambaforge3/etc/profile.d/conda.sh" ]; then
        . "/Users/YOURUSERNAME/mambaforge3/etc/profile.d/conda.sh"
    else
        export PATH="/Users/YOURUSERNAME/mambaforge3/bin:$PATH"
    fi
fi
unset __conda_setup

if [ -f "/Users/YOURUSERNAME/mambaforge3/etc/profile.d/mamba.sh" ]; then
    . "/Users/YOURUSERNAME/mambaforge3/etc/profile.d/mamba.sh"
fi
# <<< conda initialize <<<
```

And then source the file manually, see what happens:
```bash
❯  source mamba_init
(base)  ❯ 
```

**Reply by zaheen1258** [reply]
Thank you so much @milljm for your elaborate reply. I think that creating the necessary ~/.zshrc manually is the best option. However, I solved it with a different way. Yesterday before getting your response, I installed Conda with mambaforge installer from GitHub rather than following mambaforge3 installation instruction from INL getting started page. I let the installer initialise conda and it created the necessary .zshrc file for me. After that I could install MOOSE smoothly and even ran the step01 of DarcyThermoMech tutorial. Now the question is, am I going to face any problem in the future for using mambaforge instead of mambaforge3? Thanks again for all your help.

**Reply by milljm** [reply]
Do you recall the URL you installed? Our instructions point to the [Miniforge GitHub](https://github.com/conda-forge/miniforge) page. But that site has several variants. Perhaps the one you choose may be the "better" one. I'd like to try it out  for myself, and possibly change our instructions if it "just works better"!

**Reply by zaheen1258** [reply]
Yes! I do remember. I went to the following URL directly from google.

https://github.com/conda-forge/miniforge#mambaforge

and then I downloaded the installer for the apple silicon chip (Screenshot attached). Thank you.
<img width="1512" alt="Screenshot 2023-06-05 at 7 33 53 PM" src="https://github.com/idaholab/moose/assets/124659479/368c4d4f-62ce-4792-9c31-d6df3835dfbf">

**Reply by milljm** [reply]
Well, odd... That is the "group" we are instructing folks to download using `curl`, so I am not sure why the second attempt worked for you... But I am glad you got it sorted!

**Reply by zaheen1258** [reply]
Thanks a lot. Strange!... but I can think of a possibility. Before trying for 2nd time I tried creating a .zshrc file and pasting your code manually as per your direction present in these comments. However, it didn't work that time. So, I think the installer got a .zshrc file to manipulate for the 2nd time which wasn't simply present during 1st attempt. I'm not sure myself as I'm shooting in the dark here. However, this is the only difference between my 1st and 2nd attempt. Also, if I remember correctly, the cURL command did not initialise conda automatically. But during 2nd time the installer gave me a prompt in the terminal, something like this:

do you want the installer to initialise conda?(Y/n):

I think that made the difference.
